File: EXE201_Heureux/src/main/java/com/example/exe201_heureux/entity/ClassUser.java
package com.example.exe201_heureux.entity;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "Class_User")
public class ClassUser {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.DETACH)
    @JoinColumn(name = "classid")
    private Class classid;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.DETACH)
    @JoinColumn(name = "userid")
    private User userid;

    @Column(name = "flag")
    private Boolean flag;

}

File: Apis/Domain/Entities/ClassUser.cs
using System;
using System.Collections.Generic;

namespace Domain.Entities;

public class ClassUser : BaseEntity
{
    public int? Classid { get; set; }

    public ulong? Flag { get; set; }

    public int? Userid { get; set; }

    public virtual Class? Class { get; set; }

    public virtual User? User { get; set; }
}
